test/sing_for_needs_web/schema/query/causes_test.exs

```diff
@@ -181,4 +181,92 @@ defmodule SingForNeeds.Schema.Query.CauseTest do
 
     assert expected_result == json_response(conn, 200)
   end
+
+  test "causes ending soon and limit passed to causes query" do
+    causes_ending_soon_with_limit_query = """
+      query($scope: String, $limit: Int) {
+          causes(scope: $scope, limit: $limit) {
+              name
+          }
+      }
+    """
+
+    twenty_days_from_now = Timex.add(Timex.now(), Timex.Duration.from_days(20))
+    thirty_days_from_now = Timex.add(Timex.now(), Timex.Duration.from_days(30))
+    fifteen_days_from_now = Timex.add(Timex.now(), Timex.Duration.from_days(15))
+    five_days_ago = Timex.add(Timex.now(), Timex.Duration.from_days(-5))
+    nine_days_ago = Timex.add(Timex.now(), Timex.Duration.from_days(-9))
+
+    insert(:cause, %{name: "Cause ends in 30 days", end_date: thirty_days_from_now})
+    insert(:cause, %{name: "Cause ends in 20 days", end_date: twenty_days_from_now})
+    insert(:cause, %{name: "Cause ends in 15 days", end_date: fifteen_days_from_now})
+    insert(:cause, %{name: "Cause ended 5 days ago", end_date: five_days_ago})
+    insert(:cause, %{name: "Cause ended 9 days ago", end_date: nine_days_ago})
+
+    conn = build_conn()
+
+    conn =
+      post conn, "/api",
+        query: causes_ending_soon_with_limit_query,
+        variables: %{scope: "ending_soon", limit: 2}
+
+    expected_result = %{
+      "data" => %{
+        "causes" => [
+          %{"name" => "Cause ends in 15 days"},
+          %{"name" => "Cause ends in 20 days"}
+        ]
+      }
+    }
+
+    assert expected_result == json_response(conn, 200)
+  end
+
+  test "scope: trending and limit:2 passed to causes query" do
+    trending_causes_with_limit_query = """
+      query($scope: String, $limit: Int) {
+          causes(scope: $scope, limit: $limit) {
+              name
+          }
+      }
+    """
+
+    artists = insert_list(4, :artist)
+
+    insert(:cause, %{
+      name: "Cause with Medium Donation",
+      amount_raised: 30_000,
+      artists: Enum.take(artists, 2),
+      description: Faker.Lorem.paragraph()
+    })
+
+    insert(:cause, %{name: "Cause with Least Donation", amount_raised: 10_000})
+
+    insert(:cause, %{
+      name: "Cause with Least Donation and two artists",
+      amount_raised: 10_000,
+      artists: Enum.take(artists, -2)
+    })
+
+    insert(:cause, %{name: "Cause with Highest Donation", amount_raised: 90_000, artists: artists})
+
+    conn = build_conn()
+
+    conn =
```

Comment:
See note below about [my fallible opinions on] local variable naming, an extremely helpful thing for the test reader and extremely inexpensive thing to change

conn_after_trending_causes_query? Little wordy, but something to clarify what it is and avoid mutating 253's conn, which it can be argued in Elixir feels a little unidiomatic.

Reply:
I think phoenix conventionally would want us to call the conn object as just conn, as I have seen in many projects I tend to borrow from. maybe a little confusing if named different as other people may not get to know its a conn object just by the first look? what do you think?

Reply:
For the first `conn` by all means. For the second conn, `conn_after_trending_causes_query` makes the state of that conn struct a lot clearer (albeit verbosely). Just naming it conn doesn't provide the same kind of help to a test reader. 

I'm not saying you should definitely change it -- I'm bringing to your attention something that I would probably do.

Reply:
Using a different name for the conn object throws the error below 
```
** (RuntimeError) expected connection to have a response but no response was set/sent.
     Please verify that you assign to "conn" after a request:
```
I think its a convention to call in `conn`